Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServer.kt

```diff
@@ -1,12 +1,70 @@
 package de.rki.coronawarnapp.covidcertificate.signature.core.server
 
+import androidx.annotation.VisibleForTesting
+import de.rki.coronawarnapp.covidcertificate.signature.core.common.exception.DscValidationException
+import de.rki.coronawarnapp.covidcertificate.signature.core.common.exception.DscValidationException.ErrorCode
+import de.rki.coronawarnapp.server.protocols.internal.dgc.DscListOuterClass.DscList
+import de.rki.coronawarnapp.util.ZipHelper.readIntoMap
+import de.rki.coronawarnapp.util.ZipHelper.unzip
+import de.rki.coronawarnapp.util.security.SignatureValidation
+import okhttp3.ResponseBody
+import retrofit2.HttpException
+import retrofit2.Response
+import timber.log.Timber
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class DscServer @Inject constructor() {
+class DscServer @Inject constructor(
+    private val signatureValidation: SignatureValidation,
+    private val dscApi: DscApiV1
+) {
 
-    suspend fun getDscList(): ByteArray {
-        throw NotImplementedError()
+    suspend fun getDscList(): DscList {
+        return try {
+            dscApi.dscList().parseAndValidate(
+                ErrorCode.FILE_MISSING,
+                ErrorCode.SIGNATURE_INVALID,
+                ErrorCode.EXTRACTION_FAILED
```

Comment:
The error codes can be hardcoded.
This was passed in a different implementation as the parsing logic was re-used.
I think here the `dscList()` will be the only use, right?

Reply:
updated